Core.ApplicationServices/SystemUsage/ItSystemUsageService.cs

```diff
@@ -287,6 +291,84 @@ public ItSystemUsage GetById(int usageId)
             return Modify(systemUsageId, usage => usage.AddArchivePeriod(startDate, endDate, archiveId, approved));
         }
 
+        public IEnumerable<ItSystemUsage> GetSystemsByResponsibleUnitId(int unitId)
+        {
+            return _usageRepository.AsQueryable().Where(x => x.ResponsibleUsage.OrganizationUnitId == unitId).ToList();
+        }
+
+        public IEnumerable<ItSystemUsage> GetSystemsByRelevantUnitId(int unitId)
+        {
+            return _usageRepository.AsQueryable().Where(x => x.UsedBy.Select(usedBy => usedBy.OrganizationUnitId).Contains(unitId)).ToList();
+        }
+
+        public Result<ItSystemUsage, OperationError> TransferResponsibleUsage(int targetUnitId, int id)
+        {
+            return Modify(id, system =>
+            {
+                var targetUnit = system.UsedBy.FirstOrDefault(x => x.OrganizationUnitId == targetUnitId);
+                if (targetUnit == null)
+                {
+                    targetUnit = new ItSystemUsageOrgUnitUsage
+                    {
+                        ItSystemUsageId = system.Id,
+                        OrganizationUnitId = targetUnitId
+                    };
+                    system.UsedBy.Add(targetUnit);
+                }
+
+                system.ResponsibleUsage = targetUnit;
+                return Result<ItSystemUsage, OperationError>.Success(system);
+            });
+        }
+
+        public Result<ItSystemUsage, OperationError> TransferRelevantUsage(int unitId, int targetUnitId, int id)
+        {
+            return Modify(id, system =>
+            {
+                var unit = system.UsedBy.FirstOrDefault(x => x.OrganizationUnitId == unitId);
+                if (unit != null)
+                {
+                    system.UsedBy.Remove(unit);
+                }
+
+                var targetUnit = system.UsedBy.FirstOrDefault(x => x.OrganizationUnitId == targetUnitId);
+                if (targetUnit != null) 
+                    return Result<ItSystemUsage, OperationError>.Success(system);
+
+                targetUnit = new ItSystemUsageOrgUnitUsage
+                {
+                    ItSystemUsageId = system.Id,
+                    OrganizationUnitId = targetUnitId
+                };
+                system.UsedBy.Add(targetUnit);
+                return Result<ItSystemUsage, OperationError>.Success(system);
+            });
+        }
+
+        public Result<ItSystemUsage, OperationError> RemoveResponsibleUsage(int id)
+        {
+            return Modify(id, system =>
+            {
+                // WARNING: force loading so setting it to null will be tracked
+                var forceLoad = system.ResponsibleUsage;
+                system.ResponsibleUsage = null;
```

Comment:
used SystemUsageController as a template, I'm not sure if that's a right way to do it